### ISSUE TITLE
BufferQueue has been abandoned!

### ISSUE THREAD

**Issue by volser** [brianwernick/ExoMedia#206]
- [x] I have verified there are no duplicate active or recent bugs, questions, or requests

```
E/BufferQueue: [unnamed-26871-6] dequeueBuffer: BufferQueue has been abandoned!
E/ACodec: dequeueBuffer failed.
A/ACodec: frameworks/av/media/libstagefright/ACodec.cpp:3323 CHECK_EQ( mCodec->submitOutputMetaDataBuffer(),(status_t)OK) failed: -1004 vs. 0
A/libc: Fatal signal 6 (SIGABRT) at 0x000068f7 (code=-6), thread 1988 (CodecLooper)
```

###### Include the following:
 - ExoMedia version: `snapshot`
 - Device OS version: `4.4`
 - Devide Manufacturer: `Xiaomi`
 - Device Name: `Mi4`
 
###### Reproduction Steps
release view with player

###### Expected Result
no crash

###### Actual Result
crash

**Comment by volser** [comment]
E/BufferQueue: [unnamed-2053-10] queueBuffer: BufferQueue has been abandoned!
E/Surface: queueBuffer: error queuing buffer to SurfaceTexture, -19
E/MediaCodec: Codec reported an error. (omx error 0x80001001, internalError -19)
E/ExoPlayerImplInternal: Internal runtime error.
                                                                        java.lang.IllegalStateException
                                                                            at android.media.MediaCodec.dequeueOutputBuffer(Native Method)
                                                                            at com.google.android.exoplayer.MediaCodecTrackRenderer.drainOutputBuffer(MediaCodecTrackRenderer.java:889)
                                                                            at com.google.android.exoplayer.MediaCodecTrackRenderer.doSomeWork(MediaCodecTrackRenderer.java:512)
                                                                            at com.google.android.exoplayer.SampleSourceTrackRenderer.doSomeWork(SampleSourceTrackRenderer.java:129)
                                                                            at com.google.android.exoplayer.ExoPlayerImplInternal.doSomeWork(ExoPlayerImplInternal.java:434)
                                                                            at com.google.android.exoplayer.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:213)
                                                                            at android.os.Handler.dispatchMessage(Handler.java:98)
                                                                            at android.os.Looper.loop(Looper.java:136)
                                                                            at android.os.HandlerThread.run(HandlerThread.java:61)
                                                                            at com.google.android.exoplayer.util.PriorityHandlerThread.run(PriorityHandlerThread.java:40)
W/BufferQueue: [unnamed-2053-10] cancelBuffer: BufferQueue has been abandoned!

**Comment by volser** [comment]
https://github.com/google/ExoPlayer/issues/1454

**Comment by volser** [comment]
May be bug here?

`@Override
        public boolean onSurfaceTextureDestroyed(SurfaceTexture surfaceTexture) {
            surfaceTexture.release();
            emExoPlayer.blockingClearSurface();

            return true;
        }`

better will be swap surfaceTexture.release and emExoPlayer.blockingClearSurface

**Comment by brianwernick** [comment]
Nice catch.  I'll update that shortly (the fix should be in the 3.0.0-preview2 release)

**Comment by brianwernick** [comment]
This is fixed in https://github.com/brianwernick/ExoMedia/commit/0f4690adfccba450b0e3feedce6ae0c80c9467f2
